website/pages/blog/announcing-cloudquery-javascript-sdk.mdx

```diff
@@ -0,0 +1,129 @@
+---
+title: Announcing the Javascript SDK for CloudQuery Plugin Development
+tag: announcement
+date: 2023/08/31
+description:
+author: michal
+---
+
+import { BlogHeader } from "../../components/BlogHeader"
+
+<BlogHeader/>
+
+We're excited to announce the first release of a JavaScript SDK for CloudQuery plugin development! This SDK provides a high-level toolkit for developing CloudQuery plugins in JavaScript.
+
+## Background
+
+CloudQuery is designed with a pluggable architecture and uses [Apache Arrow](https://arrow.apache.org/) over [gRPC](https://grpc.io/) for communication between plugins. Source and destination plugins are independent of one another, and this architecture allows plugins to be written in different languages but still communicate with one another.
+
+Originally, we only provided SDK for writing plugins in Go only, but that is changing now. Recently, we released [SDK for Python](https://www.cloudquery.io/blog/announcing-cloudquery-python-sdk), and now we are excited to announce the SDK for JavaScript as well!
+
+![cloudquery high-level architecture](/images/cloudquery-architecture.png)
+
+## Features
+
+### Plugin Server
+
+The most basic functionality provided by the JavaScript SDK is to start a gRPC plugin server that supports all the flags expected by the CloudQuery CLI. This allows you to write a plugin in JavaScript and run it using the same command line interface as any other plugin.
+
+The following example shows how to create a plugin server that runs a plugin called `MyPlugin`:
+
+```typescript
+import { createServeCommand } from '@cloudquery/plugin-sdk-javascript/plugin/serve';
+
+import { myPlugin } from './plugin.js';
+
+const main = () => {
+  createServeCommand(myPlugin()).parse();
+};
+
+main();
+```
+
+All you need to do is to implement the NewClientFunction interface and return it.
+
+### Client Interface
+
+When implementing a source plugin, the core functionality should be handled by the three functions required by the `SourceClient` interface: `tables`, and `sync`.
+
+The `tables` function should return a list of tables that the plugin supports. The `sync` function is called when a table needs to be synced, and this is where the SDK scheduler can be used to manage the syncing of all the supported tables.
+
+There are other functions that the `Client` needs to support. Unless you need a custom implementation, you can use the `newPlugin` function to provide the basic implementation.
+
+Check out our [Airtable](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/airtable/src/plugin.ts) plugin to get an example.
+
+### Managed Asynchronous Scheduler
+
+The scheduler's main responsibilities are to manage concurrent execution of requests and the order in which tables are synced to avoid dependency issues. It also places limits on the number of concurrent requests and memory usage.
```

Comment:
```suggestion
The scheduler's main responsibilities are to manage concurrent execution of requests and the order in which tables are synced to avoid dependency issues. It also supports placing limits on the number of concurrent requests and memory usage.
```